cmake/common.cmake

```diff
@@ -0,0 +1,163 @@
+include(GNUInstallDirs)
+include(CMakePackageConfigHelpers)
+
+add_compile_options(-std=c++17)
+
+# Specific compile optios across all targets
+#add_compile_definitions(MINIMAL_LOGGING)
+
+IF(NOT CMAKE_BUILD_TYPE)
+  SET(CMAKE_BUILD_TYPE Release)
+ENDIF()
+
+# Turn off output data dumps for testing by default
+OPTION(EDGEAI_ENABLE_OUTPUT_FOR_TEST "Enable Output Dumps for test" OFF)
+
+# Check if we got an option from command line
+if(EDGEAI_ENABLE_OUTPUT_FOR_TEST)
+    message("EDGEAI_ENABLE_OUTPUT_FOR_TEST enabled")
+    add_definitions(-DEDGEAI_ENABLE_OUTPUT_FOR_TEST)
+endif()
+
+message(STATUS "CMAKE_BUILD_TYPE = ${CMAKE_BUILD_TYPE} PROJECT_NAME = ${PROJECT_NAME}")
+
+SET(CMAKE_FIND_LIBRARY_PREFIXES "" "lib")
+SET(CMAKE_FIND_LIBRARY_SUFFIXES ".a" ".lib" ".so")
+
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR}/lib/${CMAKE_BUILD_TYPE})
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR}/bin/${CMAKE_BUILD_TYPE})
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR}/bin/${CMAKE_BUILD_TYPE})
+
+if (NOT DEFINED ENV{SOC})
+    message(FATAL_ERROR "SOC not defined.")
+endif()
+
+set(TARGET_SOC_LOWER $ENV{SOC})
+
+if ("${TARGET_SOC_LOWER}" STREQUAL "j721e")
+    set(TARGET_PLATFORM     J7)
+    set(TARGET_CPU          A72)
+    set(TARGET_OS           LINUX)
+    set(TARGET_SOC          J721E)
+elseif ("${TARGET_SOC_LOWER}" STREQUAL "j721s2")
+    set(TARGET_PLATFORM     J7)
+    set(TARGET_CPU          A72)
+    set(TARGET_OS           LINUX)
+    set(TARGET_SOC          J721S2)
+elseif ("${TARGET_SOC_LOWER}" STREQUAL "j784s4")
+    set(TARGET_PLATFORM     J7)
+    set(TARGET_CPU          A72)
+    set(TARGET_OS           LINUX)
+    set(TARGET_SOC          J784S4)
+elseif ("${TARGET_SOC_LOWER}" STREQUAL "am62a")
+    set(TARGET_PLATFORM     SITARA)
+    set(TARGET_CPU          A53)
+    set(TARGET_OS           LINUX)
+    set(TARGET_SOC          AM62A)
+else()
+    message(FATAL_ERROR "SOC ${TARGET_SOC_LOWER} is not supported.")
+endif()
+
+message("SOC=${TARGET_SOC_LOWER}")
+
+add_definitions(
+    -DTARGET_CPU=${TARGET_CPU}
+    -DTARGET_OS=${TARGET_OS}
+    -DSOC_${TARGET_SOC}
+)
+
+set(DLPACK_INSTALL_DIR ${TARGET_FS}/usr/include/dlpack)
+
+link_directories(${TARGET_FS}/usr/lib/aarch64-linux-gnu
+                 ${TARGET_FS}/usr/lib/
+                 )
+
+#message("PROJECT_SOURCE_DIR =" ${PROJECT_SOURCE_DIR})
```

Comment:
@s-jain1 Are these lines supposed to be commented out?

Reply:
I have taken reference for this file from edgeai-gst-apps. And there they have commented out this line similarly.